Add playback speed adjustment for Android Auto

KeyboardHammer opened the pull request:
Adds a button to adjust the play speed from the Android Auto media player.  It's not a 1-to-1 match with the current android app play speed settings, but I did copy the existing preset options show in the android dialog.  

Also save the speed setting so it stays set between plays

![image](https://user-images.githubusercontent.com/61483659/216862929-23340fde-5c22-4162-b977-bab71803e0eb.png)

Implements #333

advplyr commented:
Great, thanks! There are a few things remaining that I couldn't knock out quickly.

1. The app allows for setting custom playback rates anything between 0.5 and 10 incremented by 0.1. We could just round to the nearest icon so if you have playback speed set to `0.7` it would show as `0.5` in Android Auto.
2. Floating points are stored with trailing decimals and I'm not sure the best way to approach that. When it comes to comparisons Kotlin is handling that fine, so this could just be as simple as rounding it before displaying. Not sure yet, need to think on it. (Screenshot below)
3. When the playback speed is updated in the app it should reflect in Android Auto. I don't think this is a requirement for this PR but thought it worth mentioning. We would need to refresh those custom actions in Android Auto which I'm not sure about.

![image](https://user-images.githubusercontent.com/67830747/217110227-bfb559b6-953f-45d1-8fba-f6968bb3a6cc.png)

KeyboardHammer commented:
Thanks for adding the event listener, it's been years since I've done android development (pre kotlin) so I didn't realize it was so simple now.

1. I added some ranges to both the icon display and the speed change so it's a little more fluid if a custom range is set.  I hesitate to add any more steps or custom ranges to Android auto so it doesn't become a distraction while driving. Not to mention anything higher than 2 is unusable for me :).
2. I fixed the conversion issue before save so it shouldn't need to be handled anywhere else, although I'm curious why the phone app doesn't have that issue saving values like 1.2f.
3. Yeah that's a nice to have but the idea is you shouldn't have to use the phone at all while driving so it's an edge case that doesn't really need to be solved.

advplyr commented:
I agree that we shouldn't add a bunch of playback speed options in android auto. This is going to have people posting bug reports and bringing it up in reviews, but I'm not sure of a better way. I think I'll check out how some other apps handle it.

I was able to implement the speed refreshing in android auto when you update it in the app. I also found a previous bug that may have caused other playback speed issues. It was setting the `userSettingsPlaybackRate` when updated in the app.

I updated the rounding for showing the icons. I'm not sure if there is a reason or you did that intentionally but I changed it so that it rounds to the nearest instead of ceiling. e.g. 2.1f was showing as `x3` but I changed it to show `x2`.
